pkg/cloudprovider/providers/azure/azure_routes.go

```diff
@@ -0,0 +1,189 @@
+func (az *AzureCloud) ListRoutes(clusterName string) (routes []*cloudprovider.Route, err error) {
+	glog.Infof("list: START clusterName=%q", clusterName)
+	glog.Infof("list: getting the route table. routeTableName=%q", az.RouteTableName)
+	routeTable, err := az.RouteTablesClient.Get(az.ResourceGroup, az.RouteTableName, "")
+	if existsRouteTable, err := checkResourceExistsFromError(err); err != nil {
+		return nil, err
+	} else if !existsRouteTable {
+		glog.Infof("list: routing table didn't exist. routeTableName=%q", az.RouteTableName)
+		return []*cloudprovider.Route{}, nil
+	}
+
+	var kubeRoutes []*cloudprovider.Route
+	if routeTable.Properties.Routes != nil {
+		kubeRoutes = make([]*cloudprovider.Route, len(*routeTable.Properties.Routes))
+		for i, route := range *routeTable.Properties.Routes {
+			instance := getInstanceName(*route.Name)
+			cidr := *route.Properties.AddressPrefix
+			glog.Infof("list: * instance=%q, cidr=%q", instance, cidr)
+
+			kubeRoutes[i] = &cloudprovider.Route{
+				Name:            *route.Name,
+				TargetInstance:  instance,
+				DestinationCIDR: cidr,
+			}
+		}
+	} else {
+		kubeRoutes = []*cloudprovider.Route{}
+	}
+
+	glog.Info("list: FINISH")
+	return kubeRoutes, nil
+}
+
+// CreateRoute creates the described managed route
+// route.Name will be ignored, although the cloud-provider may use nameHint
+// to create a more user-meaningful name.
+func (az *AzureCloud) CreateRoute(clusterName string, nameHint string, kubeRoute *cloudprovider.Route) error {
+	glog.Infof("create: clusterName=%q instance=%q cidr=%q", clusterName, kubeRoute.TargetInstance, kubeRoute.DestinationCIDR)
+
+	glog.Infof("create: getting the routetable. routeTableName=%q", az.RouteTableName)
+	routeTable, err := az.RouteTablesClient.Get(az.ResourceGroup, az.RouteTableName, "")
+	if existsRouteTable, err := checkResourceExistsFromError(err); err != nil {
+		return err
+	} else if !existsRouteTable {
+		glog.Infof("create: routetable needs creation. routeTableName=%q", az.RouteTableName)
+		routeTable = network.RouteTable{
+			Name:       to.StringPtr(az.RouteTableName),
+			Location:   to.StringPtr(az.Location),
+			Properties: &network.RouteTablePropertiesFormat{},
+		}
+
+		_, err = az.RouteTablesClient.CreateOrUpdate(az.ResourceGroup, az.RouteTableName, routeTable, nil)
+		if err != nil {
+			return err
+		}
+
+		routeTable, err = az.RouteTablesClient.Get(az.ResourceGroup, az.RouteTableName, "")
+		if err != nil {
+			return err
+		}
+	}
+
+	// ensure the subnet is properly configured
+	glog.Infof("create: getting the subnet. vnet=%q subnet=%q", az.VnetName, az.SubnetName)
+	subnet, err := az.SubnetsClient.Get(az.ResourceGroup, az.VnetName, az.SubnetName, "")
+	if existsSubnet, err := checkResourceExistsFromError(err); err != nil {
+		return err
+	} else if !existsSubnet {
+		glog.Infof("create: subnet was unexpectedly nil! vnet=%q subnet=%q", az.VnetName, az.SubnetName)
```

Comment:
Yeah, just realized glog has `Warning`. I assumed there was no `Warn` level at all since Warnf/Warnln are absent. I'll change to `Warning()`.